Documenting the usage of return type for programmatic navigation (#1480)
vaadin/docs: articles/routing/updating-url-parameters.asciidoc

---
title: Updating URL Parameters without Navigation

> description: To preserve deep linking developer must sometimes maintain the Route parameters and Query parameters manually

Reviewer: ```suggestion
description: To preserve deep linking, the developer sometimes needs to maintain the Route parameters and Query parameters manually
```

order: 25
layout: page
---

== Updating URL Parameters without Navigation

> Route parameters, Route templates and Query Parameter are most often defined in a link pointing to a specific view with the desired state and when navigating to view the Router calls the view's [methodname]`setParameter` method.

Reviewer: ```suggestion
Route parameters, Route templates and Query parameters are most often defined in a link pointing to a specific view with the desired state, and when navigating to a view, the Router calls the view's [methodname]`setParameter()` method.
```

In many cases, the view itself should also maintain the route and query parameters so that the user can share the URL from the browser's location bar meaningfully.

> A good example of such a use case could be a search view that maintains the search parameters in the URL but doesn't want to build its UI logic around the Router usage.

Reviewer: ```suggestion
A good example of such a use case could be a search view that maintains the search parameters in the URL but does not want to build its UI logic around the Router usage.
```

> Another case would be passing a more complex object to a view via Java API instead of serializing the object somehow into the URL and passing the value via Router.

Reviewer: ```suggestion
Another case would be passing a more complex object to a view via a Java API instead of serializing the object somehow into the URL and passing the value via Router.
```
